packages/ng/libraries/core/src/lib/overlay/modal/modal-panel.component.ts

```diff
@@ -51,13 +52,26 @@ export abstract class ALuModalPanelComponent<T extends ILuModalContent = ILuModa
 		this.dispose();
 	}
 	dismiss() {
-		this._ref.close();
+		if (!this._componentInstance.cancelAction) {
+			return this._ref.close();
+		}
+		const result$ = this._componentInstance.cancelAction();
+		if (result$ instanceof Observable) {
+			this.dismissClass$.next('is-loading');
+			result$.subscribe(

```

Comment:
If `cancelAction` is not correctly implemented, it could never complete or emit multiple events. A `.pipe(first())` can prevent this. (same for `submit` method)